src/main/java/use_case/CreateComment/CreateCommentInteractor.java

```diff
@@ -0,0 +1,55 @@
+package use_case.CreateComment;
+
+import entity.*;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+/**
+ * The Creation Comment Interactor.
+ */
+public class CreateCommentInteractor implements CreateCommentInputBoundary {
+    private final CreateCommentDataAccessInterface commentDataAccessObject;
+    private final CreateCommentOutputBoundary commentPresenter;
+    private final CommentFactory commentFactory;
+
+    public CreateCommentInteractor(CreateCommentDataAccessInterface commentDataAccessObject,
+                                   CreateCommentOutputBoundary commentPresenter,
+                                   CommentFactory commentFactory) {
+        this.commentDataAccessObject = commentDataAccessObject;
+        this.commentPresenter = commentPresenter;
+        this.commentFactory = commentFactory;
+    }
+
+    @Override
+    public void createComment(CreateCommentInputData createCommentInputData) {
+        Comment comment = commentFactory.createComment(
+                createCommentInputData.getAuthor(),
+                createCommentInputData.getContent(),
+                createCommentInputData.getAttachmentPath(),
+                createCommentInputData.getFileType()
+        );
+
+        // TODO :will have to fix later because you will need to access database
+        commentDataAccessObject.save(comment);
+
+        CreateCommentOutputData createCommentOutputData = new CreateCommentOutputData(
+                comment.getEntryID(),
+                comment.getAuthor(),
```

Comment:
this is a change on my end that I haven't pushed, but I think it makes more sense to pass the userID that created this comment rather than the entire user object. You could choose to do it in this PR (the program may not build due to this, but i might push my change tmrw in the lab and u could do it then) or in another one